### DISCUSSION TITLE
Have a built in way for decoding ids as part of the `inputType` function

### DISCUSSION THREAD

**Discussion by lkuoch** [StellateHQ/fuse#160] [Ideas]
Currently with fuse, every ID that gets returned back is a base64 encoded string

This can be annoying when you are doing a mutation and sending a ID field as part of the mutation payload

To solve this, I'm writing a custom `resolveID` function that tries to extract the id part if its a base64 string otherwise returns the original id.

As a result, this needs to be called everytime there is a drizzle update/insert occurs which is annoying

#### Example code
```ts
addMutationFields((t) => ({
  create_post: t.field({
    type: PostNode,
    description: "Create a new post",
    args: {
      input: t.arg({
        required: true,
        type: inputType({
          name: "CreatePostInput",
          fields: (t) => ({
            user_id: t.string({ required: true }),
            title: t.string({ required: true }),
            content: t.string({ required: true }),
          }),
        }),
      }),
    },
    resolve: (_, { input: { user_id, content, title } }) =>
      db
        .insert(Schema.Posts)
        .values({
          // Calling `resolveId` here
          user_id: resolveID(user_id),
          content,
          title,
        })
        .returning()
        .then((res) => res[0]),
  })
}));
```

I'm not sure how hard it is to change the typings as lots of the fuse code builds ontop of pothos but one option could be adding a 2nd argument config object to `inputType` eg:

```ts
inputType({
  name: "CreatePostInput",
  fields: (t) => ({
    user_id: t.string({ required: true }),
    title: t.string({ required: true }),
    content: t.string({ required: true }),
  }),
}, { nodeIDFields: ['user_id'] })
```

this option would auto resolve ID fields

**Comment by JoviDeCroock** [comment]
With `t.id` that should be taken care of, does that not work?

**Comment by lkuoch** [comment]
@JoviDeCroock TIL, thanks for this 💡

One issue I'm seeing is that when user_id becomes `t.Id()`, in the resolve the argument gets inferred as `string | number`

Is there a way to narrow it down to the `string` type?
Right now I'm calling the `to_string()` method but wondering if there's a better way

**Reply by JoviDeCroock** [reply]
That's a good point, in the future we should allow tweaking this, there is an existing discussion on this with custom scalars in https://github.com/StellateHQ/fuse/issues/17
